numpy/lib/function_base.py

```diff
@@ -689,16 +699,38 @@ def histogram(a, bins=10, range=None, normed=False, weights=None,
                 b = a[keep]
 
         if b.size == 0:
-            bins = 1
+            n_equal_bins = 1
         else:
             # Do not call selectors on empty arrays
-            width = _hist_bin_selectors[bins](b)
+            width = _hist_bin_selectors[bin_name](b)
             if width:
-                bins = int(np.ceil((mx - mn) / width))
+                n_equal_bins = int(np.ceil((mx - mn) / width))
             else:
                 # Width can be zero for some estimators, e.g. FD when
                 # the IQR of the data is zero.
-                bins = 1
+                n_equal_bins = 1
+
+    elif np.ndim(bins) == 0:
+        try:
+            n_equal_bins = operator.index(bins)
+        except TypeError:
+            raise TypeError('If a scalar, `bins` must be an integer')
+        if n_equal_bins < 1:
+            raise ValueError('If an integer, `bins` must be positive')
+
+    elif np.ndim(bins) == 1:
+        bin_edges = np.asarray(bins)
+        if np.any(bin_edges[:-1] > bin_edges[1:]):
+            raise ValueError('`bins` must increase monotonically')
```

Comment:
Moved from below - previously this was needlessly checked for a `n_bins` input if the `weight`s were weird.

Reply:
Looks like 'non-decreasing', to monotonically increasing.

Reply:
Message wasn't touched by this CR, but I guess I may as well fix it while I'm here. Can you suggest the replacement message?

Reply:
NVM, I checked, and monotonic is the correct word.